EIPS/eip-remove_utf8_support.md

```diff
@@ -0,0 +1,140 @@
+
+## Motivation
+
+UTF-8 is not a good string format for contracts, for several reasons:
+
+1. It uses variable length characters, which means you have to index the strings before using them, or iterate over parts of the string when you want to get the length, or get a character by index.
+2. It has a non-trivial string validation scheme (http://www.unicode.org/versions/Unicode10.0.0/UnicodeStandard-10.0.pdf, Table 3-7, p 126, Well-Formed UTF-8 Byte Sequences).
+3. Strings that aren't written in plain English require more bytes, which means they incur a higher gas cost.
+
+These problems has lead to a marked lack of support for string processing and (runtime) validation in contract code.
+
+The performance issues of UTF-8 is the worst part, but runtime validation is a problem as well. Most contracts are called from dedicated APIs, or web-based client UIs (such as parity), which means that validation is most likely being done at some point, but contracts can also be transacted to directly which means that no guarantees can be made unless the validation is done in contract code.
+
+The author has been working with strings in Solidity and LLL, and has written a lot of code, tests, and benchmarks related to the processing of UTF-8 and ASCII strings. An example of a library used for UTF-8 validation in Solidity can be found here: https://github.com/ethereum/solidity-examples/blob/master/docs/strings/Strings.md. Working with string processing is what motivated this EIP.
+
+## Specification
+
+In the Ethereum ABI specification, a `string` is a "dynamically sized unicode string assumed to be UTF-8 encoded". Strings are currently encoded in the same way as `bytes`, with the same 32 byte header. The header contains the "length" of the string, which is the number of bytes that the string occupies.
+
+The new string header would instead have three parts: *length*, *range*, and *reserved*.
+
+| Section              | position (little endian)  |  size  | total size |
+|----------------------|---------------------------|--------|------------|
+| length (bytes)       | 0                         | 8      | 8          |
+| range                | 8                         | 8      | 16         |
+| reserved             | 16                        | 16     | 32         |
+
+##### length
+
+The length is the length of the data in bytes (same as it is now).
+
+##### range
+
+The range is an unsigned 8 byte integer, which is to be interpreted as the upper bound for character values. The lower bound is always `0`.
+
+A range of `0` means that the standard byte encoding is used, so it is the same as setting it to `255`. This is to make `string` compatible with `bytes`.
+
+Validating a string is done by checking that none of its characters has a value that is outside of the range.
+
+Note that non power-of-two character-widths are not ideal, because they lead to inefficient use of storage (a single value could have its bytes spread out over more then one storage slot).
```

Comment:
Perhaps the header should also contain the character widths?

Reply:
yes, extra bytes is essentially the width, or "byte width", of a character minus one. i give more answer in the other question about extra bytes. 

"range" btw is the maximum value that a character may assume. the minimum value is always assumed to be 0, so does not have to be included.